CMakeLists.txt

```diff
@@ -289,7 +289,7 @@ endif()
 # Add appropriate flags for GCC
 if (LLVM_COMPILER_IS_GCC_COMPATIBLE)
   set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11 -fno-common -Woverloaded-virtual -Wcast-qual -fno-strict-aliasing -pedantic -Wno-long-long -Wall -W -Wno-unused-parameter -Wwrite-strings")
-  option(WITH_WERROR "Compile with '-Werror' C compiler flag" ON)
+  option(WITH_WERROR "Compile with '-Werror' C compiler flag" OFF)
```

Comment:
Do these changes cause the compiler to emit warnings?

Reply:
In file included from /builds-1/steleman/llvm-git/20180614-flang-github-cavium-60/llvm-git/tools/flang/runtime/libpgmath/lib/aarch64/acos.c:18:0:
/builds-1/steleman/llvm-git/20180614-flang-github-cavium-60/llvm-git/tools/flang/runtime/include/mthdecls.h:552:24: warning: ISO C does not allow extra ‘;’ outside of a function [-Wpedantic]
 FLTDECL_C(__mth_i_cabs);
                        ^
In file included from /builds-1/steleman/llvm-git/20180614-flang-github-cavium-60/llvm-git/tools/flang/runtime/libpgmath/lib/aarch64/acosh.c:21:0:
/builds-1/steleman/llvm-git/20180614-flang-github-cavium-60/llvm-git/tools/flang/runtime/include/mthdecls.h:552:24: warning: ISO C does not allow extra ‘;’ outside of a function [-Wpedantic]
 FLTDECL_C(__mth_i_cabs);
                        ^
/builds-1/steleman/llvm-git/20180614-flang-github-cavium-60/llvm-git/tools/flang/runtime/include/mthdecls.h:553:27: warning: ISO C does not allow extra ‘;’ outside of a function [-Wpedantic]
 CMPLXDECL_C(__mth_i_cacos);
                           ^
/builds-1/steleman/llvm-git/20180614-flang-github-cavium-60/llvm-git/tools/flang/runtime/include/mthdecls.h:554:27: warning: ISO C does not allow extra ‘;’ outside of a function [-Wpedantic]
 CMPLXDECL_C(__mth_i_casin);

[ ... etc etc etc ... ]

Reply:
Would you remove the warning from the new and changed files please?  We'll get after the existing files with warning.

Reply:
There are **many** more other warnings throughout flang code, unrelated to libpgmath:

> /builds-1/steleman/llvm-git/20180625-flang-github-cavium-60/llvm-git/tools/flang/tools/flang2/flang2exe/aarch64-Linux/ll_abi.c: In function ‘update_homogeneous’:
/builds-1/steleman/llvm-git/20180625-flang-github-cavium-60/llvm-git/tools/flang/tools/flang2/flang2exe/aarch64-Linux/ll_abi.c:64:3: warning: enumeration value ‘DT_NONE’ not handled in switch [-Wswitch]
>   switch (dtype) {
>   ^~~~~~

> /builds-1/steleman/llvm-git/20180625-flang-github-cavium-60/llvm-git/tools/flang/tools/flang2/flang2exe/aarch64-Linux/ll_abi.c:64:3: warning: enumeration value ‘DT_WORD’ not handled in switch [-Wswitch]
> /builds-1/steleman/llvm-git/20180625-flang-github-cavium-60/llvm-git/tools/flang/tools/flang2/flang2exe/aarch64-Linux/ll_abi.c:64:3: warning: enumeration value ‘DT_DWORD’ not handled in switch [-Wswitch]
> /builds-1/steleman/llvm-git/20180625-flang-github-cavium-60/llvm-git/tools/flang/tools/flang2/flang2exe/aarch64-Linux/ll_abi.c:64:3: warning: enumeration value ‘DT_HOLL’ not handled in switch [-Wswitch]
> /builds-1/steleman/llvm-git/20180625-flang-github-cavium-60/llvm-git/tools/flang/tools/flang2/flang2exe/aarch64-Linux/ll_abi.c:64:3: warning: enumeration value ‘DT_BINT’ not handled in switch [-Wswitch]
> /builds-1/steleman/llvm-git/20180625-flang-github-cavium-60/llvm-git/tools/flang/tools/flang2/flang2exe/aarch64-Linux/ll_abi.c:64:3: warning: enumeration value ‘DT_SINT’ not handled in switch [-Wswitch]

[ ... etc etc etc ... ]

> /tools/flang2/flang2exe/aarch64-Linux/ll_abi.c:73:10: warning: implicit declaration of function ‘size_of’ [-Wimplicit-function-declaration]
>    size = size_of(dtype);
>           ^~~~~~~

[ ... etc etc etc ... ]

> /builds-1/steleman/llvm-git/20180625-flang-github-cavium-60/llvm-git/tools/flang/tools/flang2/flang2exe/aarch64-Linux/ll_abi.c:231:7: warning: implicit declaration of function ‘alignment’; did you mean ‘indent’? [-Wimplicit-function-declaration]
>    if (alignment(dtype) > 4 && size % 8 == 0) {
>        ^~~~~~~~~
>        indent

[ ... etc etc etc ... ]
> /builds-1/steleman/llvm-git/20180625-flang-github-cavium-60/llvm-git/tools/flang/tools/flang2/flang2exe/bihutil.c:46:22: warning: passing argument 4 of ‘stg_alloc’ discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
>    STG_ALLOC(bihb, 128);
>                       ^
[ ... etc etc etc ... ]

Yes, I agree, flang should compile with zero warnings. That would make -Werror possible.

There are **numerous** const-correctness warnings throughout flang.

This is GCC 7.3. Note: I am **not** compiling with -ansi. That would make GCC very angry on all the K&R C function definitions.

Reply:
I have placed the full output from building flang - with warnings - on T99 AArch64 here:

https://drive.google.com/open?id=19XCxUYDO39iih3zKQPEO5CqPHC2-oHoA

Warning: file is 28MB in size.

Reply:
I don't get those warnings w/ gcc 7.3.0.  Seems like -pedantic isn't passed through for me.

Reply:
What does *Seems like -pedantic isn't passed through for me* mean?

Is -pedantic passed on compile-line?

Many of the warnings triggered by -pedantic are **serious** and they deserve attention.

Reply:
*Seems like* means "it appears to be true but not known for sure" -- in this case, I don't see pedantic on the verbose command line but I did not investigate why pedantic does not appear to appear for me.